Crash when address already bound

Instead of a graceful exit the scraper panics with a segfault if the listening address is already bound

```
level=error timestamp=2024-06-12T01:21:25.773821082Z caller=utils.go:76 msg="http server failed: listen tcp :9630: bind: address already in use"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x88f2d5]

goroutine 1 [running]:
database/sql.(*DB).Close(0x0)
	/home/craig/apps/go-1.22/src/database/sql/sql.go:910 +0x35
github.com/Vonng/pg_exporter/exporter.(*Exporter).Close(0xc000228000)
	/home/craig/projects/EDB/pg_exporter/exporter/exporter.go:209 +0x188
github.com/Vonng/pg_exporter/exporter.Run()
	/home/craig/projects/EDB/pg_exporter/exporter/main.go:219 +0xd17
main.main()
	/home/craig/projects/EDB/pg_exporter/main.go:22 +0xf
```

--- a/exporter/exporter.go
+++ b/exporter/exporter.go
@@ -206,9 +206,11 @@ func (e *Exporter) Close() {
 	}
 	// close peripheral servers (we may skip acquire lock here)
 	for _, srv := range e.IterateServer() {
-		err := srv.Close()
-		if err != nil {
-			logErrorf("fail closing server %s: %s", e.server.Name(), err.Error())
+		if srv == nil {
+			err := srv.Close()
+			if err != nil {
+				logErrorf("fail closing server %s: %s", e.server.Name(), err.Error())
+			}
 		}
 	}
 	logInfof("pg exporter closed")

